tests/Integration/Embeddings/VectorStores/Elasticsearch/ElasticsearchVectorStoreTest.php

```diff
@@ -27,7 +27,7 @@
     $embeddingQuery = json_decode($rawFileContent, true);
 
     $client = (new ClientBuilder())::create()
-        ->setHosts(['http://localhost:9200'])
+        ->setHosts([getenv('ELASTIC_URL') ?? 'http://elasticsearch:9200'])
```

Comment:
it's not localhost ?

Reply:
I'm running PHP inside a docker container spinned up along with redis, pgvector and elasticsearch:
```
services:
    php:
        container_name: ${COMPOSE_PROJECT_NAME}-php
        stdin_open: true
        tty: true
        build:
            context: php
            args:
                PHP_VERSION: ${PHP_VERSION}
                COMPOSE_PROJECT_NAME: ${COMPOSE_PROJECT_NAME}
        extra_hosts:
            - "host.docker.internal:host-gateway"
        volumes:
            - ${APP_PATH}:/var/${COMPOSE_PROJECT_NAME}
        environment:
            PHP_IDE_CONFIG: serverName=PHPSTORM
            ELASTIC_URL: ${ELASTIC_URL}
            PGVECTOR_HOST: pgvector
            REDIS_HOST: redis
            MILVUS_HOST: milvus

    elasticsearch:
        image: docker.elastic.co/elasticsearch/elasticsearch:8.13.4
        container_name: ${COMPOSE_PROJECT_NAME}-elasticsearch
        environment:
            - discovery.type=single-node
            - xpack.security.enabled=false
            - node.name=${COMPOSE_PROJECT_NAME}-elasticsearch
            - cluster.name=es-cluster
            - cluster.routing.allocation.disk.threshold_enabled=false
            - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
        ports:
            - "9200:9200"
        volumes:
            - esdata:/usr/share/elasticsearch/data

    pgvector:
      # In production, you may want to use a managed database service
      image: ankane/pgvector
      container_name: ${COMPOSE_PROJECT_NAME}-pgvector
      environment:
        - POSTGRES_DB=postgres
        - POSTGRES_USER=myuser
        # You should definitely change the password in production
        - POSTGRES_PASSWORD=!ChangeMe!
      volumes:
        - ./pgvector/scripts:/docker-entrypoint-initdb.d
        - db-data:/var/lib/postgresql/data:rw
        # You may use a bind-mounted host directory instead, so that it is harder to accidentally remove the volume and lose all your data!
        # - ./docker/db/data:/var/lib/postgresql/data:rw
      ports:
        - '5432:5432'

    redis:
      image: redis/redis-stack:latest
      ports:
        - 6379:6379
        - 8001:8001
volumes:
    esdata:
    db-data: { }
```
Whith this configuration PHP machine can see other services with these names instead of localhost:
```
            ELASTIC_URL: http://elasticsearch:9200
            PGVECTOR_HOST: pgvector
            REDIS_HOST: redis
            MILVUS_HOST: milvus
```

Reply:
can we find a solution that works for both when we run the tests outside your container and inside. 
If you let localhost with the port forwarding 9200:9200 it should work no?